gammapy/spectrum/make.py

```diff
@@ -195,3 +195,120 @@ def run(self, observation, selection=None):
             kwargs["edisp"] = self.make_edisp(observation)
 
         return SpectrumDataset(**kwargs)
+
+
+class SafeMaskMaker:
+    """Make safe data range mask for a given observation.
+
+    Parameters
+    ----------
+    methods : {"aeff-default", "aeff-max", "edisp-bias"}
+        Method to use for the safe energy range. Can be a
+        list with a combination of those. Resulting masks
+        are combined with logical `and`. "aeff-default"
+        uses the energy ranged specified in the DL3 data
+        files, if available.
+    aeff_percent : float
+        Percentage of the maximal effective area to be used
+        as lower energy threshold for method "aeff-max".
+    bias_percent : float
+        Percentage of the energy bias to be used as lower
+        energy threshold for method "edisp-bias"
+    """
+
+    def __init__(
+        self, methods=["aeff-default"], aeff_percent=10, bias_percent=10
```

Comment:
Don't put mutable contains like lists or dicts as default arguments. If needed, put `None` and put the default in the method. https://docs.python-guide.org/writing/gotchas/